tourist_scheduling_system/src/agents/autonomous_guide_agent.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env python3
+# Copyright AGNTCY Contributors (https://github.com/agntcy)
+# SPDX-License-Identifier: Apache-2.0
+"""
+Autonomous Guide Agent - A2A Client with OpenAI LLM
+
+An intelligent tour guide agent that uses OpenAI to make autonomous decisions
+about availability, pricing, and specialties. It continuously operates and
+responds to market conditions.
+"""
+
+import asyncio
+import json
+import logging
+import os
+import random
+from datetime import datetime, timedelta
+from typing import Dict, List
+
+import click
+import httpx
+try:
+    from openai import AsyncAzureOpenAI  # type: ignore
+except Exception:  # pragma: no cover - openai optional for demo
+    AsyncAzureOpenAI = None  # fallback sentinel
+from a2a.client import ClientFactory, ClientConfig
+from a2a.client.client_factory import minimal_agent_card
+from a2a.client.helpers import create_text_message_object
+try:
+    from dotenv import load_dotenv  # type: ignore
+except Exception:  # pragma: no cover
+    def load_dotenv():  # fallback no-op
+        return None
```

Comment:
pydantic settings is kind of the defacto for these kind of settings and doing validating them etc e.g.

```
class AgentConfig(pydantic_settings.BaseSettings):
	azure_openai_deployment_name: str
	azure_openai_api_key: str
    ...
```